tests/exercise_v2.ts

```diff
@@ -0,0 +1,527 @@
+      await provider.connection.requestAirdrop(
+        payer.publicKey,
+        100 * LAMPORTS_PER_SOL
+      ),
+      "confirmed"
+    );
+    await provider.connection.confirmTransaction(
+      await provider.connection.requestAirdrop(
+        minter.publicKey,
+        100 * LAMPORTS_PER_SOL
+      ),
+      "confirmed"
+    );
+    await provider.connection.confirmTransaction(
+      await provider.connection.requestAirdrop(
+        exerciser.publicKey,
+        100 * LAMPORTS_PER_SOL
+      ),
+      "confirmed"
+    );
+  });
+
+  describe("Non-nft OptionMarket", () => {
+    before(async () => {
+      // Initialize a new OptionMarket
+      ({
+        quoteToken,
+        underlyingToken,
+        optionToken,
+        underlyingAmountPerContract,
+        quoteAmountPerContract,
+        optionMarketKey,
+        exerciseFeeKey,
+        optionMarket,
+        remainingAccounts,
+        instructions,
+      } = await initSetup(provider, payer, mintAuthority, program));
+      await initOptionMarket(
+        program,
+        payer,
+        optionMarket,
+        remainingAccounts,
+        instructions
+      );
+      // Create a new minter
+      const {
+        optionAccount: minterOptionAcct,
+        underlyingAccount: minterUnderlyingAccount,
+        writerTokenAccount: minterWriterAcct,
+      } = await createMinter(
+        provider.connection,
+        minter,
+        mintAuthority,
+        underlyingToken,
+        new anchor.BN(100)
+          .mul(optionMarket.underlyingAmountPerContract)
+          .muln(2)
+          .toNumber(),
+        optionMarket.optionMint,
+        optionMarket.writerTokenMint,
+        quoteToken
+      );
+      // Mint a bunch of contracts to the minter
+      const { ix: mintOptionsIx } =
+        await psyAmericanInstructions.mintOptionV2Instruction(
+          minterProgram,
+          minterOptionAcct.publicKey,
+          minterWriterAcct.publicKey,
+          minterUnderlyingAccount.publicKey,
+          new anchor.BN(100),
+          optionMarket
+        );
+      await program.provider.send(new Transaction().add(mintOptionsIx), [
+        minter,
+      ]);
+      // Create an exerciser
+      ({
+        optionAccount: exerciserOptionAcct,
+        quoteAccount: exerciserQuoteAcct,
+        underlyingAccount: exerciserUnderlyingAcct,
+      } = await createExerciser(
+        provider.connection,
+        exerciser,
+        mintAuthority,
+        quoteToken,
+        new anchor.BN(100)
+          .mul(optionMarket.quoteAmountPerContract)
+          .muln(2)
+          .toNumber(),
+        optionMarket.optionMint,
+        underlyingToken.publicKey
+      ));
+
+      // Transfer a options to the exerciser
+      await optionToken.transfer(
+        minterOptionAcct.publicKey,
+        exerciserOptionAcct.publicKey,
+        minter,
+        [],
+        new u64(100)
+      );
+    });
+    beforeEach(async () => {
+      size = new u64(2);
```

Comment:
nit: constant could be moved out of the beforeEach and made a const within the describe block

Reply:
or am i mistaken and it gets mutated when we call the arithmetic functions on it